docs/reST/tut/newbieguide.rst

```diff
@@ -123,93 +127,89 @@ output file had the same pixel format as the input file.  If you're writing a
-
-A final note -- not every game requires high framerates. A strategic wargame
-could easily get by on just a few updates per second -- in this case, the added
-complexity of dirty rect animation may not be necessary.
-
-
-There is NO rule six.
----------------------
-
-
-Hardware surfaces are more trouble than they're worth.
-------------------------------------------------------
-
-**Especially in pygame 2, because HWSURFACE now does nothing**
+Even though this technique is not required for making performant 2D games with
+modern CPUs, it is still useful to be aware of. There are also still plenty of other ways
+to accidentally tank your game's performance with poorly optimized rendering logic.
+For example, even on modern hardware it's probably too slow to call ``set_at`` once per pixel
+on the display surface. Being mindful of performance is still something you'll have to
+do.
 
-If you've been looking at the various flags you can use with
-``pygame.display.set_mode()``, you may have thought like this: `Hey,
-HWSURFACE! Well, I want that -- who doesn't like hardware acceleration. Ooo...
-DOUBLEBUF; well, that sounds fast, I guess I want that too!`.  It's not
-your fault; we've been trained by years of 3-d gaming to believe that hardware
-acceleration is good, and software rendering is slow.
+There just aren't that many 'one neat trick to fix your code performance' tips. Every game
+is different and there are different problems and different algorithms to solve them
+efficiently in each type of game. Pretty much every time your 2D game code is failing to hit a
+reasonable frame rate the underlying cause turns out to be bad algorithm or a misunderstanding
+of fundamental game design patterns.
 
-Unfortunately, hardware rendering comes with a long list of drawbacks:
+If you are having performance problems, first make sure you aren't loading files repeatedly in your
+game loop, then use one of the many options for profiling your code to find out what is taking up the
+most time. Once you are armed with at least some knowledge on why your game is slow, try asking the
+internet (via google), or the pygame community if they've got some better algorithms to help you out.
 
- * It only works on some platforms. Windows machines can usually get hardware surfaces if you ask for them. Most other platforms can't. Linux, for example, may be able to provide a hardware surface if X4 is installed, if DGA2 is working properly, and if the moons are aligned correctly. If a hardware surface is unavailable, SDL will silently give you a software surface instead.
+**HWSURFACE and DOUBLEBUF**
 
- * It only works fullscreen.
+These :func:`.display.set_mode()` flags do nothing in pygame versions 2.0.0 and
+later (you can check the docs if you don't believe me)!  There's no reason to
+use them anymore.  Even in pygame 1, their effects were pretty nuanced and
+generally misunderstood by most pygame users.  They were never magic speed-up
+flags, unfortunately.
```

Comment:
Sadly, I've been wrong about this. Doublebuf still has a use for OpenGL stuff. I've gotta open a PR to correct the docs where I said doublebuf is obsolete.

However, this is right that it is not a magic speed up flag!

Reply:
:) I adjusted this section to reflect truth and this sentiment.